javascript/401-code-challenges/hashmap-leftJoin/index.js

```diff
@@ -0,0 +1,50 @@
+'use strict';
+
+
+// function leftJoin(synonyms, antonyms) {
+//   // Check if hashmap is empty
+//   if (Object.keys(synonyms).length === 0 ||
+//   Object.keys(antonyms).length === 0) {
+//     return null;
+//   }
+
+//   let result = {};
+
+//   for (let key in synonyms) {
+//     let synonymsValue = synonyms[key];
+//     let antonymsValue = antonyms[key] || null;
+//     result[key] = [synonymsValue, antonymsValue];
+//   }
+
+//   return result;
+// }
+
+
+// module.exports = { leftJoin };
+
+function join(synonyms, antonyms, joinType = 'LEFT') {
+  // Check if hashmap is empty
+  if (Object.keys(synonyms).length === 0 || Object.keys(antonyms).length === 0) {
+    return null;
+  }
```

Comment:
We only want to return return null if the left map is empty. If the right map is empty we will return the expected map with a null in the right position for every row. (Obviously, it's all flipped for a right join).